src/redisgears_python.c

```diff
@@ -1758,19 +1758,28 @@ static PyObject* executeCommand(PyObject *cls, PyObject *args){
     }
     const char* commandStr = PyUnicode_AsUTF8AndSize(command, NULL);
 
-    RedisModuleString** argements = array_new(RedisModuleString*, 10);
+    RedisModuleString** arguments = array_new(RedisModuleString*, 10);
     for(int i = 1 ; i < PyTuple_Size(args) ; ++i){
-        PyObject* argument = PyTuple_GetItem(args, i);
-        PyObject* argumentStr = PyObject_Str(argument);
         size_t argLen;
-        const char* argumentCStr = PyUnicode_AsUTF8AndSize(argumentStr, &argLen);
+        const char* argumentCStr = NULL;
+        PyObject* argument = PyTuple_GetItem(args, i);
+        if(PyByteArray_Check(argument)) {
+            argLen = PyByteArray_Size(argument);
+            argumentCStr = PyByteArray_AsString(argument);
+        } else if(PyBytes_Check(argument)) {
+            argLen = PyBytes_Size(argument);
+            argumentCStr = PyBytes_AsString(argument);
+        } else {
+            PyObject* argumentStr = PyObject_Str(argument);
+            argumentCStr = PyUnicode_AsUTF8AndSize(argumentStr, &argLen);
+            Py_DECREF(argumentStr);
```

Comment:
The decrease ref must come after you use the underline pointer. Otherwise there is a potential issue that the pointer might be freed on decreasing ref count..